discord/ui/section.py

```diff
@@ -0,0 +1,230 @@
+        self._underlying.components.clear()
+        for item in items:
+            self._add_component_from_item(item)
+
+    def add_item(self, item: Item) -> None:
+        """Adds an item to the section.
+
+        Parameters
+        ----------
+        item: :class:`Item`
+            The item to add to the section.
+
+        Raises
+        ------
+        TypeError
+            An :class:`Item` was not passed.
+        ValueError
+            Maximum number of items has been exceeded (10).
+        """
+
+        if len(self.items) >= 3:
+            raise ValueError("maximum number of children exceeded")
+
+        if not isinstance(item, Item):
+            raise TypeError(f"expected Item not {item.__class__!r}")
+
+        self.items.append(item)
+        self._add_component_from_item(item)
+
+    def get_item(self, id: int) -> Item | None:
+        """Get an item from this section. Alias for `utils.get(section.items, id=id)`.
+
+        Parameters
+        ----------
+        id: :class:`int`
+            The id of the item to get
+
+        Returns
+        -------
+        Optional[:class:`Item`]
+            The item with the matching ``id`` if it exists.
+        """
+        if not id:
+            return None
+        if self.accessory and self.accessory.id == id:
+            return self.accessory
+        return get(self.items, id=id)
+
+    def add_text(self, content: str, *, id: int | None = None) -> None:
+        """Adds a :class:`TextDisplay` to the section.
+
+        Parameters
+        ----------
+        content: :class:`str`
+            The content of the text display.
+        id: Optiona[:class:`int`]
+            The text display's ID.
+
+        Raises
+        ------
+        ValueError
+            Maximum number of items has been exceeded (3).
+        """
+
+        if len(self.items) >= 3:
+            raise ValueError("maximum number of children exceeded")
+
+        text = TextDisplay(content, id=id)
+
+        self.add_item(text)
+
+    def set_accessory(self, item: Item) -> None:
+        """Set an item as the section's :attr:`accessory`.
+
+        Parameters
+        ----------
+        item: :class:`Item`
+            The item to set as accessory.
+            Currently only supports :class:`~discord.ui.Button` and :class:`~discord.ui.Thumbnail`.
+
+        Raises
+        ------
+        TypeError
+            An :class:`Item` was not passed.
+        """
+
+        if not isinstance(item, Item):
+            raise TypeError(f"expected Item not {item.__class__!r}")
+        if self.view:
+            item._view = self.view
+
+        self.accessory = item
+        self._underlying.accessory = item._underlying
+
+    def set_thumbnail(
+        self,
+        url: str,
+        *,
+        description: str | None = None,
+        spoiler: bool = False,
+        id: int | None = None,
+    ) -> None:
+        """Set a :class:`Thumbnail` with the provided url as the section's :attr:`accessory`.

```

Comment:
```suggestion
        """Sets a :class:`Thumbnail` with the provided URL as the section's :attr:`accessory`.
```
Subject-predicate agreement; the function is singular